main.tf

```diff
@@ -60,3 +60,28 @@ resource "azurerm_automation_account" "pre-aa" {
 
   tags = var.common_tags
 }
+
+data "azurerm_key_vault_secret" "slack_monitoring_address" {
+  name         = "slack-monitoring-address"
+  key_vault_id = data.azurerm_key_vault.keyvault.id
+}
+
+resource "azurerm_application_insights" "this" {
+  count               = var.env == "prod" ? 1 : 0
+  name                = "pre-${var.env}-appinsights"
+  location            = data.azurerm_resource_group.rg.location
+  resource_group_name = data.azurerm_resource_group.rg.name
+  application_type    = "other"
+}
+
```

Comment:
Can we add the instrumentation key to the KV too please? need it in the API app

```
resource "azurerm_key_vault_secret" "AZURE_APPINSIGHTS_KEY" {
  name         = "AppInsightsInstrumentationKey"
  value        = azurerm_application_insights.appinsights.instrumentation_key
  key_vault_id = module.key-vault.key_vault_id
}
```

Reply:
sure can, will raise this in a separate pr